api/base/serializers.py

```diff
@@ -328,6 +350,26 @@ def __init__(self, **kwargs):
         self.link_type = kwargs.pop('link_type', 'url')
         super(JSONAPIHyperlinkedGuidRelatedField, self).__init__(read_only=True, **kwargs)
 
+    def resolve(self, resource):
+        """
+        Resolves the view for target node or target comment when embedding.
+        """
+        embed_value = resource.target._id
+        if resource.target._name == 'node':
+            view_name = 'nodes:node-detail'
+            lookup_url_kwarg = 'node_id'
+        else:
```

Comment:
I'm not sure it's a good idea to fall back to comments here; what if `resource.target._name` isn't `node` or `comments`? It's probably best to raise an error.